Tests/BasicTests/JSONTests.swift

```diff
@@ -26,6 +26,21 @@ class JSONTests: XCTestCase {
         XCTAssertEqual(encode(.array([.int(1), .string("hi")])), "[1, \"hi\"]")
         XCTAssertEqual(encode(.dictionary(["a": .int(1), "b": .string("hi")])), "{\"a\": 1, \"b\": \"hi\"}")
     }
+
+    func testEncodingWithContentsKeyOrder() {
+        // Test encoding with customized contents key order.
```

Comment:
Can you write a test that uses something other than the specific keys above? For instance, something that shows that the predicate I suggested is correct but wouldn't work with your original predicate?